Pokedex/ViewController.swift

```diff
@@ -0,0 +1,345 @@
+        
+        var indecies : [Int] = []
+        var rand: Int
+        var i: Int = 1
+        while i < 21 {
+            rand = Int(arc4random_uniform(UInt32(pokemon.count)))
+            //                if !searchedPokemon.contains(pokemon[rand]) {
+            //                    searchedPokemon.append(pokemon[rand])
+            //                    i += 1
+            //                }
+            if !indecies.contains(rand) {
+                indecies.append(rand)
+                i += 1
+                searchedPokemon.append(pokemon[rand])
+            }
+        }
+        searchBar.text = ""
+        pressed = nil
+        nameButton.backgroundColor = UIColor.black
+        numButton.backgroundColor = UIColor.black
+        attackButton.backgroundColor = UIColor.black
+        healthButton.backgroundColor = UIColor.black
+        defenseButton.backgroundColor = UIColor.black
+        randomButton.backgroundColor = UIColor.black
+        performSegue(withIdentifier: "toCollection", sender: self)
+    }
+    
+    func buttonPressed(b : UIButton) {
+        nameButton.backgroundColor = UIColor.black
+        numButton.backgroundColor = UIColor.black
+        attackButton.backgroundColor = UIColor.black
+        healthButton.backgroundColor = UIColor.black
+        defenseButton.backgroundColor = UIColor.black
+        randomButton.backgroundColor = UIColor.black
+        
+        b.backgroundColor = UIColor.gray
+        pressed = b
+    }
+    
+    func categoriesHooHah() {
+        if searchBar.text != "" {
+            noPokemonToBeSearched = false
+            switch pressed {
+            case nameButton:
+                for p in pokemon {
+                    if p.name == searchBar.text {
+                        searchedPokemon.append(p)
+                        break
+                    }
+                }
+            case numButton:
+                for p in pokemon {
+                    if String(p.number) == searchBar.text! {
+                        searchedPokemon.append(p)
+                        break
+                    }
+                }
+            case attackButton:
+                for p in pokemon {
+                    if String(p.attack) >= searchBar.text! {
+                        searchedPokemon.append(p)
+                    }
+                }
+            case healthButton:
+                for p in pokemon {
+                    if String(p.health) >= searchBar.text! {
+                        searchedPokemon.append(p)
+                    }
+                }
+            case defenseButton:
+                for p in pokemon {
+                    if String(p.defense) >= searchBar.text! {
+                        searchedPokemon.append(p)
+                    }
+                }
+            default:
+                break
+            }
+        } else {
+            noPokemonToBeSearched = true
+        }
+        searchBar.text = ""
+        pressed = nil
```

Comment:
The nil value for pressed is never handled. Either set a default (ie. the name button) or handle this case with an error pop-up that tells the user they have no selected something to search by.